src/analysis/fourier/fourier_transform.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import analysis.special_functions.complex.circle
+import measure_theory.group.integration
+
+/-!
+# The Fourier transform
+
+We set up the Fourier transform for complex-valued functions on finite-dimensional spaces.
+
+## Design choices
+
+We define Fourier transforms in the following context:
+* `𝕜` is a commutative ring.
+* `V` and `W` are `𝕜`-modules.
+* `e` is a unitary additive character of `𝕜`, i.e. a homomorphism `(multiplicative 𝕜) →* circle`.
+* `μ` is a measure on `V`.
+* `L` is a `𝕜`-bilinear form `V × W → 𝕜`.
+* `E` is a complete normed `ℂ`-vector space.
+
+With these definitions, the Fourier transform is a map from functions `V → E` to
+functions `W → E`, defined by sending `f` to
+
+`λ w, ∫ v in V, e [-L v w] • f v ∂μ`,
+
+where `e [x]` is notational sugar for `(e (multiplicative.of_add x) : ℂ)` (available in locale
+`fourier_transform`).
+
+The most familiar case, of course, is
+* `𝕜 = V = W = ℝ`
+* `e = fourier_char` (defined below), i.e. the character `λ x, exp (2 * π * I * x)`
+* `μ = volume`
+* `L = linear_map.mul ℝ ℝ`
+
+The notation `𝓕` is available in the locale `fourier_transform` as a shortcut for this case.
+However, we set things up much more generally (number theorists care about the case when `𝕜` is an
+adele ring, for instance).
```

Comment:
Can you mention that standard examples are when `V = W` is an inner product space and `L` is the inner product, and also when `W` is the dual of `V` and `L` is the standard pairing?